ostd/src/mm/page_table/node/mod.rs

```diff
@@ -242,209 +268,73 @@ where
 
     /// Converts the handle into a raw handle to be stored in a PTE or CPU.
     pub(super) fn into_raw(self) -> RawPageTableNode<E, C> {
         let this = ManuallyDrop::new(self);
 
-        let raw = this.page.paddr();
-
+        // Release the lock.
         this.page.meta().lock.store(0, Ordering::Release);
 
-        RawPageTableNode {
-            raw,
-            _phantom: PhantomData,
-        }
+        // SAFETY: The provided physical address is valid and the level is
+        // correct. The reference count is not changed.
+        unsafe { RawPageTableNode::from_raw_parts(this.page.paddr(), this.page.meta().level) }
     }
 
     /// Gets a raw handle while still preserving the original handle.
     pub(super) fn clone_raw(&self) -> RawPageTableNode<E, C> {
-        core::mem::forget(self.page.clone());
+        let page = ManuallyDrop::new(self.page.clone());
 
-        RawPageTableNode {
-            raw: self.page.paddr(),
-            _phantom: PhantomData,
-        }
+        // SAFETY: The provided physical address is valid and the level is
+        // correct. The reference count is increased by one.
+        unsafe { RawPageTableNode::from_raw_parts(page.paddr(), page.meta().level) }
     }
 
-    /// Gets an extra reference of the child at the given index.
-    pub(super) fn child(&self, idx: usize, in_tracked_range: bool) -> Child<E, C> {
-        debug_assert!(idx < nr_subpage_per_huge::<C>());
-
-        let pte = self.read_pte(idx);
-        if !pte.is_present() {
-            Child::None
-        } else {
-            let paddr = pte.paddr();
-            if !pte.is_last(self.level()) {
-                // SAFETY: We have a reference count to the page and can safely increase the reference
-                // count by one more.
-                unsafe {
-                    Page::<PageTablePageMeta<E, C>>::inc_ref_count(paddr);
-                }
-                Child::PageTable(RawPageTableNode {
-                    raw: paddr,
-                    _phantom: PhantomData,
-                })
-            } else if in_tracked_range {
-                // SAFETY: We have a reference count to the page and can safely
-                // increase the reference count by one more.
-                unsafe {
-                    DynPage::inc_ref_count(paddr);
-                }
-                // SAFETY: The physical address of the PTE points to a forgotten
-                // page. It is reclaimed only once.
-                Child::Page(unsafe { DynPage::from_raw(paddr) }, pte.prop())
-            } else {
-                Child::Untracked(paddr, pte.prop())
-            }
-        }
+    /// Gets the number of valid PTEs in the node.
+    pub(super) fn nr_children(&self) -> u16 {
+        // SAFETY: The lock is held so we have an exclusive access.
+        unsafe { *self.page.meta().nr_children.get() }
     }
 
-    /// Replace the child at the given index with a new child.
+    /// Read a non-owning PTE at the entry.
```

Comment:
`Read` -> `Reads`

What do you want to mean by saying "at the entry"? Perhaps you want to say "at the index"?